### PR TITLE
JDK-8286256 : Update libxml2 to 2.9.14

### PR DESCRIPTION
**Pull request by HimaBinduMeda**
Updated libxml to version 2.9.14.As mentioned in UPDATING.txt, configured for windows , linux and Mac platforms and updated the files accordingly.

Updated libxslt to version 1.1.35. Generated the config.h files for windows , Mac and linux platforms and updated accordingly.

Verified the build on all the platforms and sanity testing looks fine.No regressions observed.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 reviewer, 1 author)

### Issues
 * [JDK-8286256](https://bugs.openjdk.java.net/browse/JDK-8286256): Update libxml2 to 2.9.14
 * [JDK-8286257](https://bugs.openjdk.java.net/browse/JDK-8286257): Update libxslt to 1.1.35

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Johan Vos](https://openjdk.java.net/census#jvos) (@johanvos - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/797/head:pull/797` \
`$ git checkout pull/797`

Update a local copy of the PR: \
`$ git checkout pull/797` \
`$ git pull https://git.openjdk.java.net/jfx pull/797/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 797`

View PR using the GUI difftool: \
`$ git pr show -t 797`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/797.diff">https://git.openjdk.java.net/jfx/pull/797.diff</a>

</details>

### CODE REVIEW

**Comment by HimaBinduMeda** [comment]
Creating UPDATING.txt for libxslt. Will commit it once done.

**Review by kevinrushforth — COMMENTED** [review]
The code changes look good. I tested it on 3 platforms. All good.

I left a few inline comments, including a couple questions about whitespace diffs. I notice that there are 27 files that only have whitespace changes and no other diffs, as well as one more where the whitespace diffs are the majority of the changes. Is this intentional?

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.c [hunk 1]
```diff
@@ -64,7 +64,7 @@
 #define IS_BLANK(c) (((c) == 0x20) || ((c) == 0x09) || ((c) == 0xA) ||  \
                      ((c) == 0x0D))
 
-#define IS_BLANK_NODE(n)                        \
+#define IS_BLANK_NODE(n)                                                \
```
[the remark this answers is not in the document]
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/attributes.c:67]: Yes, like in libxml, updated indentation here as well.

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/UPDATING.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+7.1 Copy `libxslt/src/config.h` to `libxslt/mac/config.h` and follow same guidelines as Windows to retain changes from our repo.
+
+- For Linux
+8. Prerequisites to compile libxslt: libtool, autoconf, automake
+9. Configure libxslt.
+        > cd modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/src
+        > ./configure
+9.1 Copy `libxslt/src/config.h` to `libxslt/linux/config.h` and follow same guidelines as Windows to retain changes from our repo.
+
+10. Remove tabs and trailing whitespaces from source files(.h and .c).
```
[the remark this answers is not in the document]
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/UPDATING.txt:36]: Done

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxml/src/valid.c [hunk 3]
```diff
@@ -27,13 +27,13 @@
 #include <libxml/globals.h>
 
 static xmlElementPtr xmlGetDtdElementDesc2(xmlDtdPtr dtd, const xmlChar *name,
-                               int create);
+                                   int create);
```
[the remark this answers is not in the document]
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxml/src/valid.c:30]: These changes are due to the indentation corrections. Now, the formatting looks fine.
Tab conversion looks fine for all the code blocks like if, switch and functions.

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/UPDATING.txt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+2. Extract contents into 'modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/src' directory.
+3. We do not have any local changes in the libxslt source but libxslt source needs to be configured before integrating into JavaFX.
+Below are platform wise steps to configure.
+
+- For windows
+4. Configure libxslt.
+        > cd modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/src/win32
+        >  cscript configure.js compiler=msvc xslt_debug=no  iconv=no mem_debug=no modules=no  zlib=no profiler=no trio=no
+        - Above command generates a header file 'src/config.h' file (on all platforms) and libxslt\src\libxslt\xsltconfig.h.
+4.1 Copy `libxslt\src\config.h` to `libxslt\win32\config.h'. config.h file defines several macros to control libxslt features. We do not require all of the features to be enabled.
```
[the remark this answers is not in the document]
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/UPDATING.txt:15]: Done

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt [hunk 5]
```diff
@@ -41,15 +41,17 @@ Updating libxml in OpenJFX:
 
 9.2 Copy libxml\src\config.h to libxml\linux\config.h
 
-10. Helper commands for removing tabs and trailing whitespaces from source files(.h and .c).
+10. Update version info in 'modules/javafx.web/src/main/legal/libxml.md'.Also, update copyright if any new files are added.
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt:44]: Minor: missing space before `Also`.

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/UPDATING.txt [hunk 6]
```diff
@@ -33,4 +34,6 @@ Below are platform wise steps to configure.
         > ./configure
 9.1 Copy `libxslt/src/config.h` to `libxslt/linux/config.h` and follow same guidelines as Windows to retain changes from our repo.
 
-10. Remove tabs and trailing whitespaces from source files(.h and .c).
+10. Update version info in 'modules/javafx.web/src/main/legal/libxslt.md'.Also, update copyright if any new files are added.
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/UPDATING.txt:37]: Minor: missing space before `Also`.

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/linux/libexslt/exsltconfig.h [hunk 7]
```diff
@@ -18,7 +18,7 @@ extern "C" {
  *
  * the version string like "1.2.3"
  */
-#define LIBEXSLT_DOTTED_VERSION "1.1.34"
+#define LIBEXSLT_DOTTED_VERSION "0.8.20"
```
[the remark this answers is not in the document]
Comment (kevinrushforth) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/linux/libexslt/exsltconfig.h:21]: Actually, now that I see it, it looks correct as is, so you can ignore this question.

**Review by kevinrushforth — APPROVED** [review]
The updated instructions look good with one minor issue (a missing space between a `.` and the start of the next sentence).

**Comment by kevinrushforth** [comment]
@johanvos or @tiainen Can one of you be the second reviewer on this?

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/UPDATING.txt (hunk 6 above)
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxslt/UPDATING.txt]: done

File: modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt (hunk 5 above)
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/ThirdParty/libxml/UPDATING.txt]: done

**Review by johanvos — APPROVED** [review]
works with our new build approach.

**Pull request closed without merging**